spring-batch-infrastructure/src/main/java/org/springframework/batch/item/file/transform/DelimitedLineTokenizer.java

```diff
@@ -133,6 +134,8 @@ public void setQuoteCharacter(char quoteCharacter) {
 	@Override
 	protected List<String> doTokenize(String line) {
 
+		Assert.state(null != delimiter && 0 != delimiter.length());
```

Comment:
We don't want to be doing this for every record.  We also would want to know there's an issue before this point.  What about implementing `InitializerBean` and providing the check there?